pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/AvoidUsingHardCodedIPRule.java

```diff
@@ -135,8 +137,9 @@ protected boolean isIPv6(final char firstChar, String s, final boolean checkIPv6
         // Quick check before using Regular Expression
         // 1) At least 3 characters
         // 2) 1st must be a Hex number or a : (colon)
-        // 3) Must contain at least 1 : (colon)
-        if (s.length() < 3 || !(isHexCharacter(firstChar) || firstChar == ':') || s.indexOf(':') < 0) {
+        // 3) Must contain at least 2 colons (:)
+        if (s.length() < 3 || !(isHexCharacter(firstChar) || firstChar == ':') || s.indexOf(':') < 0
+                || StringUtils.countMatches(s, ':') < 2) {
```

Comment:
can't be remove `s.indexOf(':') < 0`? this seems to be more restrictive

Reply:
Yes, this can be removed. I've pushed another commit.